### ISSUE TITLE
 Optional Fields are also being expected in response.

### ISSUE THREAD

**Issue by barath121** [specmatic/specmatic#1214]
**Description**
Optional Fields are also being expected in response.

**Steps to reproduce**
1. Create a schema where keys are not required similar to this.
```yaml
DeliverySummaryResult:
      type: object
      properties:
        delivery_event_level:
          type: array
          items:
            $ref: '#/components/schemas/DeliveryEventLevelSchema'
        delivery_summary:
          $ref: '#/components/schemas/DeliverySummarySchema'
```
2. Server should respond with some empty response.
```json
{}
```
3. This error is thrown
```sh
 Reason: Testing scenario "Webhook delivery summary. Response: Success"
        API: GET /service/partner/webhook/v1.0/organization/(organization_id:string)/extension/(extension_id:string)/report/delivery_summary -> 200
  
          >> RESPONSE.BODY.delivery_event_level
          
             Value mismatch: key delivery_event_level was missing
          
          >> RESPONSE.BODY.delivery_summary
          
             Value mismatch: key delivery_summary was missing
```
**Expected behavior**
Test case should pass.

**System Information:**
 - Specmatic version: 1.3.35(Node.js Wrapper).

**Comment by harikrishnan83** [comment]
@barath121 Thanks for reporting the issue. We are taking a look.

**Comment by barath121** [comment]
@harikrishnan83 Any update on this?

**Comment by harikrishnan83** [comment]
@barath121 Thanks for following up. This is fixed in [release 2.0.7](https://github.com/znsio/specmatic/releases/tag/2.0.7). Closing the issue for now. Please feel free to re-open if you have any questions on this.
